MAINT: `pre`-tests with dev versions are not running anymore

from latest merge commit test run

https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=4558&view=logs&j=3618ddf5-2391-5260-f9eb-8f35f95d270b&t=a24edd62-b2b9-54c8-9892-98a88509521b

pytest problems with pandas or pytest config problem ?

```
Starting: Run tests (pip pre)
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.201.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/b92f0bde-a9f5-44a0-a8a5-1ef1be8202fc.sh
Testing pip-pre
pytest -m  --skip-examples --junitxml=junit/test-results.xml -n auto --dist loadscope --durations=25 statsmodels
ERROR: while parsing the following warning configuration:

  error::pandas.core.common.SettingWithCopyWarning

This error occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1642, in parse_warning_filter
    category: Type[Warning] = _resolve_warning_category(category_)
  File "/opt/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1681, in _resolve_warning_category
    cat = getattr(m, klass)
AttributeError: module 'pandas.core.common' has no attribute 'SettingWithCopyWarning'


##[error]Bash exited with code '4'.
```

--- a/setup.cfg
+++ b/setup.cfg
@@ -43,7 +43,6 @@ filterwarnings =
     error:recarray support has been deprecated:FutureWarning
     error:The value returned will change to a:FutureWarning
     error:The default value of lags:FutureWarning
-    error::pandas.core.common.SettingWithCopyWarning
     error:non-integer arg n is deprecated:DeprecationWarning
     error:Creating an ndarray:numpy.VisibleDeprecationWarning
     error:The default number of lags:FutureWarning:

--- a/statsmodels/stats/descriptivestats.py
+++ b/statsmodels/stats/descriptivestats.py
@@ -392,6 +392,9 @@ class Description:
 
         def _mode(ser):
             mode_res = stats.mode(ser.dropna())
+            # Changes in SciPy 1.10
+            if np.isscalar(mode_res[0]):
+                return float(mode_res[0]), mode_res[1]
             if mode_res[0].shape[0] > 0:
                 return [float(val) for val in mode_res]
             return np.nan, np.nan

